src/Microsoft.ApplicationInsights.AspNetCore/Implementation/ApplicationInsightsInitializer.cs

```diff
@@ -43,7 +43,7 @@ internal class ApplicationInsightsInitializer : IObserver<DiagnosticListener>, I
         /// </summary>
         public void Start()
         {
-            DiagnosticListener.AllListeners.Subscribe(this);
+            this.subscriptions.Add(DiagnosticListener.AllListeners.Subscribe(this));
```

Comment:
Seems like the magic should keep working since we are keep subscribing to the DiagnoticSource, but some how it doesn't work for that case?